repository/db/db_connection.go

```diff
@@ -21,10 +20,12 @@ func NewDBConnection() error {
 
 	db, err := gorm.Open(postgres.Open(dbConnAttrs), &gorm.Config{})
 	if err != nil {
-		return err
+		return nil, err
+	}
+
+	if db.AutoMigrate(&model.User{}) != nil {
```

Comment:
The structure of the user model and users table in DB are not the same, there is also no deleted_at and many other problems which I added in the comment within migration.

Reply:
https://gorm.io/docs/models.html They are same :D